org.eclipse.cdt.cross.arc.gnu/src/com/arc/cdt/toolchain/arc/ArcOptionEnablementManager.java

```diff
@@ -36,13 +36,14 @@
 
 public class ArcOptionEnablementManager extends OptionEnablementManager {
 
-    private static final String TCF_OPTION_ID = "org.eclipse.cdt.cross.arc.gnu.base.option.target.tcf";
-    private static final String TCF_FILE_OPTION_ID = "org.eclipse.cdt.cross.arc.gnu.base.option.target.filefortcf";
-    private static final String TCF_MEMORY_MAP = "org.eclipse.cdt.cross.arc.gnu.base.option.target.maptcf";
+    private static final String TCF_OPTION_ID = getTCF("tcf");
+    private static final String TCF_FILE_OPTION_ID = getTCF("filefortcf");
+    private static final String TCF_MEMORY_MAP = getTCF("maptcf");
+    private static final String TCF_IMPORT_C_DEFINES = getTCF("includecdefines");
```

Comment:
It is better to change those static fields to contains only the last part - "includecdefines" and define those fields as public. Then:
```java
private static final String[] TCF_RELATED_OPTIONS = { getTCF(TCF_FILE_OPTION_ID), ... };
```
Then also add a static function in this class:
```java
public static boolean tcfOptionIdMatch(id, option)
{
  return id.indexOf(".option.target." + option) > 0;
}
```
and use that to replace code like `} else if (sID.indexOf("option.target.importcdefines") > 0) {`